suite-common/bluetooth/src/bluetoothReducerCreator.ts

```diff
@@ -0,0 +1,133 @@
+import { AnyAction, Draft } from '@reduxjs/toolkit';
+
+import { createReducerWithExtraDeps } from '@suite-common/redux-utils';
+import { deviceActions } from '@suite-common/wallet-core';
+
+import {
+    bluetoothAdapterEventAction,
+    bluetoothConnectDeviceEventAction,
+    bluetoothDeviceListUpdate,
+    bluetoothScanStatusAction,
+} from './bluetoothActions';
+
+export type BluetoothScanStatus = 'idle' | 'running' | 'error';
+
+export type BluetoothDevice = {
+    id: string;
+    name: string;
+    data: number[];
+    lastUpdate: number;
+    status: DeviceBluetoothStatus | null;
+};
+
+export type DeviceBluetoothStatus =
+    | { type: 'pairing'; pin?: string }
+    | { type: 'paired' }
+    | { type: 'connecting' }
+    | { type: 'connected' }
+    | {
+          type: 'error';
+          error: string;
+      };
+
+export type DeviceBluetoothStatusType = DeviceBluetoothStatus['type'];
+
+type BluetoothState<T extends BluetoothDevice> = {
+    isAdapterEnabled: boolean;
+    scanStatus: BluetoothScanStatus;
+
+    // This will be persisted, those are devices we believed that are paired
+    // (because we already successfully paired them in the Suite) in the Operating System
+    knownDevices: T[];
+
+    // This list of devices that is union of saved-devices and device that we get from scan
+    devices: T[];
+};
+
+export const bluetoothReducerCreator = <T extends BluetoothDevice>() => {
+    const initialState: BluetoothState<T> = {
+        isAdapterEnabled: true, // To prevent the UI from flickering when the page is loaded
+        scanStatus: 'running', // To prevent the UI from flickering when the page is loaded
+        knownDevices: [] as T[],
+        devices: [] as T[],
+    };
+
+    return createReducerWithExtraDeps<BluetoothState<T>>(initialState, (builder, extra) =>
+        builder
+            .addCase(bluetoothAdapterEventAction, (state, { payload: { isPowered } }) => {
+                state.isAdapterEnabled = isPowered;
+                if (!isPowered) {
+                    state.devices = [];
+                    state.scanStatus = 'done';
```

Comment:
This seems to be a typo.
```suggestion
                    state.scanStatus = 'idle';
```